### DISCUSSION TITLE
Welcome to ADReportingTools Discussions!

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [jdhitsolutions/ADReportingTools#12] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we  build together 💪.

Please post bugs and problems in Issues.

**Comment by pedromendesmct** [comment]
Thanks for sharing! looks great , just in time to a project i am involved !

**Reply by jdhitsolutions** [reply]
Definitely interested to hear how it works for you. Or doesn't.

**Reply by pedromendesmct** [reply]
Hi, will feedback you for sure, i had some issues installing the module, i had i first run with it in a lab enviroment, and really liked to use it, will deep dive into it, right now i have to do a kind of auditng in a quite big AD infra  (4000 + users multi-forest and multi-site), your tool will for sure help me. thanks

**Reply by furicle** [reply]
Are there Powershell version requirements?   

I can't seem to install it the way suggested, and I don't have an '-AllowPreRelease' option.

Perhaps a pointer in the installation area to more documentation around installing from the gallery?

**Reply by pedromendesmct** [reply]
Hello, i had that issue also, after some tries (on 2 diferent PCs in one VM) i found thiis article https://www.c-amie.co.uk/technical/unable-to-update-nuget-or-packages-in-powershell-due-to-warning-unable-to-download-the-list-of-available-providers-check-your-internet-connection/  and the this fix  resolved the issue , something with TLS versions) and it installed in all the test PCs and run like a breeze

**Reply by jdhitsolutions** [reply]
You might need to update the PowerShellGet module. Take a look at the beginning of https://jdhitsolutions.com/blog/powershell/8027/powershell-modules-in-a-cross-version-world/ for some background and guidance. The module should work wherever the Active Directory cmdlets work.

**Reply by jdhitsolutions** [reply]
The module is no longer in pre-release version so installing it from the PowerShell Gallery should work without issue.

**Comment by slm04747** [comment]
I probably missed this in the documentation, but does this module require PS7.1? The module installs in 5.1, but when I try to import it throws up its hands in despair with an error for start-threadjob. If I run it in 7.1 all is good. I still run a lot of PS5.1 because of exchange. (I think you have said something about switching over to 7.1 as much as possible.)

**Reply by jdhitsolutions** [reply]
This module does NOT require PowerShell 7, but it does require the ThreadJob module which you can install from the PowerShell Gallery. 

```powershell
Install-Module ThreadJob
```

But this tells me I need to update the manifest to add this requirement. Thanks for catching this.

**Comment by slm04747** [comment]
Thanks for the quick response. I noticed that the ansi colors do not always reset after a command:
[Graphical user interface, text  Description automatically generated]

I’m looking forward to showing this off to the rest of my team. They’ll find it very useful.

Steve

From: Jeff Hicks ***@***.***>
Reply-To: jdhitsolutions/ADReportingTools ***@***.***>
Date: Friday, April 2, 2021 at 11:27 AM
To: jdhitsolutions/ADReportingTools ***@***.***>
Cc: Stephan Moss ***@***.***>, Comment ***@***.***>
Subject: Re: [jdhitsolutions/ADReportingTools] Welcome to ADReportingTools Discussions! (#12)


This module does NOT require PowerShell 7, but it does require the ThreadJob module which you can install from the PowerShell Gallery.

Install-Module ThreadJob

But this tells me I need to update the manifest to add this requirement. Thanks for catching this.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjdhitsolutions%2FADReportingTools%2Fdiscussions%2F12%23discussioncomment-563262&data=04%7C01%7Csteve.moss%40pomona.edu%7C247a7f3e7c2a4ed1d1da08d8f604eaaf%7C817f590439044ee8b3a5a65d4746ff70%7C0%7C0%7C637529848277883396%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=cGPD6AqZlWZm%2BAXZXO6Ld2eMfACWmB6UuCQZh6HzqGE%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABOFXULWR4KFAQGFIENCNB3TGYD7RANCNFSM4YUIMHFQ&data=04%7C01%7Csteve.moss%40pomona.edu%7C247a7f3e7c2a4ed1d1da08d8f604eaaf%7C817f590439044ee8b3a5a65d4746ff70%7C0%7C0%7C637529848277893392%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=XyRcAYImN8WIgCq9xoRZ5CGZbbV3QUkjxTC2yNS%2BnKY%3D&reserved=0>.

________________________________
[EXTERNAL EMAIL] Exercise caution before clicking on links or opening attachments.

**Comment by jdhitsolutions** [comment]
Yeah, if you look at the `$ADReportingToolOptions` variable, it doesn't use the reset ANSI sequence. Run `Get-ADReportingToolsOptions` to reset the console.

**Comment by jdhitsolutions** [comment]
I can add a reset value to the variable so that if you look at $ADReportingToolsOptions, it will reset your terminal. This will be coming in v1.3.0
